stdlib/public/core/StringGraphemeBreaking.swift

```diff
@@ -86,44 +86,50 @@ private func _hasGraphemeBreakBetween(
 private func _measureCharacterStrideICU(
   of utf8: UnsafeBufferPointer<UInt8>, startingAt i: Int
 ) -> Int {
-  let iterator = _ThreadLocalStorage.getUBreakIterator(utf8)
-  let offset = __swift_stdlib_ubrk_following(
-    iterator, Int32(truncatingIfNeeded: i))
+  // ICU will gives us a different result if we feed in the whole buffer, so
+  // slice it appropriately.
```

Comment:
We care about the behavior of String, for which we have tests. Why would we care if a detail of an ICU API that we no longer use changes?

Reply:
The value of that test would be to call out the specific difference in behaviour between 5.0 and 5.1. This has an observable effect on how indexing works, doesn't it?

Reply:
We already have that test in this PR, and it would fail on 5.0. We don't usually add test cases trying to mimic prior bugs by subverting the stdlib's interfaces.